refinery/data_set_manager/models.py

```diff
@@ -646,19 +657,28 @@ def run_generate_auxiliary_node_task(self):
                 self.file_item.filetype.used_for_visualization and
                 self.file_item.datafile and
                 settings.REFINERY_AUXILIARY_FILE_GENERATION ==
-                'on_file_import'):
+                'on_file_import' and
+                self.file_item.get_extension().lower() in
+                self.AUXILIARY_FILES_NEEDED_FOR_VISUALIZATION):
             # Create an empty FileStoreItem (we do the datafile association
             # within the generate_auxiliary_file task
             auxiliary_file_store_item = FileStoreItem.objects.create()
 
             auxiliary_node = self._create_and_associate_auxiliary_node(
                 auxiliary_file_store_item
             )
-            result = data_set_manager.tasks.generate_auxiliary_file.delay(
-                auxiliary_node, self.file_item
+            generate = data_set_manager.tasks.generate_auxiliary_file.subtask(
+                (auxiliary_node, self.file_item,)
             )
-            auxiliary_file_store_item.import_task_id = result.task_id
-            auxiliary_file_store_item.save()
+            file_import = FileImportTask().subtask(
+                (auxiliary_node.file_item.uuid, None,),
+                immutable=True
```

Comment:
what is "this" and why is it unnecessary?

Reply:
Line 675 (if a comment doesn't specify a line range then it is referring to the line directly above it) but now that you mention it, the args on line 674 would not be needed also since "the first task executes passing its return value to the next task in the chain" (http://docs.celeryproject.org/en/3.1/userguide/canvas.html#the-primitives).

Reply:
ah i see - generate_auxiliary_file does not currently do that.  were you saying you wanted that? it's not like the FileStoreItem is generated there

Reply:
Yes, the name implies that it should generate the FileStoreItem and return its UUID but there may be more refactoring required.

Reply:
ok i have done some refactoring